tree/ntuple/v7/test/CMakeLists.txt

```diff
@@ -21,12 +21,26 @@ if(MSVC)
     COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/libCustomStruct.dll
                                      ${CMAKE_CURRENT_BINARY_DIR}/libCustomStruct.dll)
 endif()
-ROOT_ADD_GTEST(ntuple ntuple.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore CustomStruct)
-ROOT_ADD_GTEST(ntuple_metrics ntuple_metrics.cxx LIBRARIES ROOTNTuple)
-ROOT_ADD_GTEST(ntuple_minifile ntuple_minifile.cxx LIBRARIES ROOTNTuple)
-ROOT_ADD_GTEST(ntuple_packing ntuple_packing.cxx LIBRARIES ROOTNTuple)
-ROOT_ADD_GTEST(ntuple_pages ntuple_pages.cxx LIBRARIES ROOTNTuple)
-ROOT_ADD_GTEST(ntuple_print ntuple_print.cxx LIBRARIES ROOTNTuple CustomStruct)
-ROOT_ADD_GTEST(ntuple_raw ntuple_raw.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore)
+
+ROOT_ADD_GTEST(ntuple_basics ntuple_basics.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore CustomStruct)
```

Comment:
In general, I think many of the new tests do not require `MathCore`, `ROOTDataFrame`, or `CustomStruct`.  Let's link against them only if necessary.

Reply:
I tried leaving them out, e.g.  
`ROOT_ADD_GTEST(ntuple_basics ntuple_basics.cxx LIBRARIES ROOTNTuple)`
but ran into some CMake errors because of all the includes in the single header file. I think CMake adjusts what headers are available to each test?
```text
Scanning dependencies of target ntuple_basics
[100%] Building CXX object tree/ntuple/v7/test/CMakeFiles/ntuple_basics.dir/ntuple_basics.cxx.o
In file included from /home/max/projects/root/tree/ntuple/v7/test/ntuple_basics.cxx:1:0:
/home/max/projects/root/tree/ntuple/v7/test/ntuple_test.hxx:5:31: fatal error: ROOT/RDataFrame.hxx: No such file or directory
 #include <ROOT/RDataFrame.hxx>
                               ^
compilation terminated.
tree/ntuple/v7/test/CMakeFiles/ntuple_basics.dir/build.make:82: recipe for target 'tree/ntuple/v7/test/CMakeFiles/ntuple_basics.dir/ntuple_basics.cxx.o' failed
make[2]: *** [tree/ntuple/v7/test/CMakeFiles/ntuple_basics.dir/ntuple_basics.cxx.o] Error 1
CMakeFiles/Makefile2:36341: recipe for target 'tree/ntuple/v7/test/CMakeFiles/ntuple_basics.dir/all' failed
```

Reply:
Ok for now, we can revisit this later.